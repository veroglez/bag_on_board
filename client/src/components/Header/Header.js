import React, {} from 'react';
import { bool, func, string } from 'prop-types';
import { useStateValue } from 'common/context/contextHooks';

// import mix from '../../common/mix';
import { Button } from 'components';
import styles from './Header.module.scss';


const Header = ({ disabled, onClick, text, ...inherit }) => {
  const [state] = useStateValue();
  const { l10n } = state;

  return (
    <div className={styles.container}>
      <Button text={l10n.ADD} />
    </div>
  );
};

Header.propTypes = {
  disabled: bool,
  icon: bool,
  onClick: func,
  text: string,
};

Header.defaultProps = {
  disabled: false,
  icon: undefined,
  onClick: undefined,
  text: undefined,
};

export default Header;
