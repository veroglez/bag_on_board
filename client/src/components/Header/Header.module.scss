@import '../../common/style.scss';

.container {
  padding: $spaceM;
  border-bottom: $border;
  display: 'flex';
  justify-content: 'space-bettwen';
}