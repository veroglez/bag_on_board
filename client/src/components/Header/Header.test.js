import React from 'react';
import renderer from 'react-test-renderer';
import * as Context from 'common/context/contextHooks';
import Header from './Header';

const STATE = {
  l10n: { ADD: 'add' },
  cards: [
    { name: 'Vero', bags: 2 },
  ],
};

const HOC = () => (
  <Context.StateContext.Provider value={[STATE]}>
    <Header />
  </Context.StateContext.Provider>
);

describe('<Header>', () => {
  it('renders', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
