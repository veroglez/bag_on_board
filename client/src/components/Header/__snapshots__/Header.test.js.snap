// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders 1`] = `
<div
  className="container"
>
  <button
    className={
      Array [
        "button",
      ]
    }
    disabled={false}
    type="submit"
  >
    <span>
      add
    </span>
  </button>
</div>
`;
